kafka-0.10.x/src/main/scala/monix/kafka/KafkaProducerConfig.scala

```diff
@@ -270,6 +271,9 @@ case class KafkaProducerConfig(
     "metrics.sample.window.ms" -> metricsSampleWindow.toMillis.toString
   )
 
+  def toJavaMap: java.util.Map[String, Object] =
+    toMap.filter(_._2 != null).mapValues(_.asInstanceOf[AnyRef]).asJava
```

Comment:
Is filtering `null` necessary? I believe optional are only `ssl` related configs and I think at least some of them accept null as valid value

Reply:
I get a `NullPointerException` for the partitioner when sending a message, unless I manage the partition value myself.  This does not happen when using the properties version of the config and narrowed it down to this one difference.